mpp/include/mpp_domains_misc.inc

```diff
@@ -413,12 +413,13 @@ end subroutine init_nonblock_type
     if(any(pelist2 == pe)) then
        call mpp_get_data_domain(domain2, isd, ied, jsd, jed)
        call mpp_get_compute_domain(domain2, is, ie, js, je)
-       allocate(field1(isd:ied, jsd:jed),field2(isd:ied, jsd:jed))
+       allocate(field2(isd:ied, jsd:jed))
       if((size(field_in,1) .ne. ied-isd+1) .or. (size(field_in,2) .ne. jed-jsd+1)) &
          call mpp_error(FATAL,'mpp_check_field: input field is not on the data domain')
       field2(isd:ied, jsd:jed) = field_in(:,:)
     endif
 
+    allocate(field1(isd:ied, jsd:jed))
```

Comment:
isd, ied, jsd, jed are only given values for pe's in pelist2.  Suggest putting in an if-test for pe's in pelist1 and allocating field1 to either (1,1) or domain1's index extents.

Also, need to ensure field1 will be properly deallocated for the pe's on which it will now be allocated.

Reply:
allocating field1 to (1,1) worked well for pelist1.